### PR TITLE
Rename install.sh to installer.sh and Add Uninstall

### LINKED ISSUE #1733: Uninstall Script
[kind/Feature]

As far as I can tell, there is no uninstall script available for `odo` at this time. It would be helpful to have an OS-independent script similar to the install script. 

## Which functionality do you think we should add?
Uninstall `odo` in an OS-independent manner.

## Why is this needed?
Give users the ability to uninstall `odo`. My use case is that I would like to remove an old version from an environment I am working in.

### PR DESCRIPTION
**Pull request by danielhelfand**
## What is the purpose of this change? What does it change?
This pull request edits the `install.sh` os-independent script for macOS/Linux for installing `odo` to also uninstall `odo`. Since the script now handles both installation and uninstall, the script has been renamed `installer.sh`. 

The assumption made by these changes is that the user will have installed `odo` via this script as well.

To install `odo`, users will be able to continue using the command in the `README` for `odo`, but will need to change the name to `installer.sh`: 

```
curl -L https://github.com/openshift/odo/raw/master/scripts/installer.sh | bash
```

To uninstall `odo`, users can run the following command:

```
curl -L https://github.com/openshift/odo/raw/master/scripts/installer.sh --uninstall | bash
```

Currently, the script would only have the ability to support one flag (i.e., `--uninstall`). Ideally, the script should be able to handle multiple flags, but I'm not sure of the best way to do this in bash, so I welcome any suggestions with regard to that. 

Assuming this is an acceptable approach, I can edit the `README` to include uninstall instructions.

## Was the change discussed in an issue?
fixes #1733 

## How to test changes?
1. Run `installer.sh` on the supported operating systems (i.e., macOS and Linux distributions) to install `odo`. Verify `odo` has been installed correctly.
2. Run `installer.sh --uninstall` on the supported operating systems (i.e., macOS and Linux distributions) after installation of `odo` and verify `odo` has been properly uninstalled. 

I have also edited the tests associated with this script to test for uninstalling `odo`. Because of the name change to `installer.sh`, I also changed the test script to `test-installer.sh`.

### CODE REVIEW

File: scripts/installer.sh [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+                echo "# Downloading odo from $BINARY_URL"
+                curl -Lo $TMP_DIR/odo "$BINARY_URL"
+            ;;
+
+            latest)
+                BINARY_URL="$GITHUB_RELEASES_URL/odo-$platform.gz"
+                echo "# Downloading odo from $BINARY_URL"
+                curl -Lo $TMP_DIR/odo.gz "$BINARY_URL"
+                echo "# Extracting odo.gz"
+                gunzip -d $TMP_DIR/odo.gz
```
Comment (danielhelfand) [inline scripts/installer.sh]: A note that we may want to think of a way to check if `gunzip` is locally installed. Ran into issues with this.

File: scripts/test-installer.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        echo "******************************************************"
+        FAILED_UNINSTALL="$FAILED_UNINSTALL $image"
+    fi
+    echo ""
+done
+
+if [ -n "$FAILED_INSTALL" ] || [ -n "$FAILED_UNINSTALL" ]; then
+    echo "TEST FAILED!!"
+    echo "Installation script failed in following images:"
+    echo "Intsall test failures: $FAILED_INSTALL"
```
[the remark this answers is not in the document]
Comment (danielhelfand) [reply scripts/test-installer.sh]: Thanks for catching that. Pushed up a change.

**Comment by surajnarwade** [comment]
@danielhelfand , please make change here also: https://github.com/openshift/odo/blob/master/scripts/bump-version.sh#L29

**Comment by danielhelfand** [comment]
@surajnarwade Sure thing and thanks! Just changed it.

**Comment by mohammedzee1000** [comment]
We need to figure out a way to run that test script automatically in CI. Other than that, looks good

**Comment by amitkrout** [comment]
@mohammedzee1000 Why there is no travis job run ?

**Comment by mohammedzee1000** [comment]
The last run of the tests was in May. It's possible it ran while travis was disabled and never ran after

**Comment by amitkrout** [comment]
@danielhelfand can you please rebase to make sure travis job won't make any issue

**Comment by danielhelfand** [comment]
@amitkrout Sure thing. Just went ahead and did it.

**Comment by cdrage** [comment]
@danielhelfand Something funky is happening with rebase. Mind rebasing again? Shows 27 commits in this PR.

**Comment by danielhelfand** [comment]
@cdrage Think I have gotten into a strange state, so I'm going to close this pr and fork again. I'll open up a fresh pr. Sorry for the troubles here.

**Comment by danielhelfand** [comment]
This has been reopened as #1802

**Pull request closed without merging**